src/GovUk.Education.ExploreEducationStatistics.Data.Services.Tests/ReleaseServiceTests.cs

```diff
@@ -243,6 +244,139 @@ await statisticsDbContext.IndicatorGroup.AddRangeAsync(
+            );
+
+            var result = await service.ListSubjects(contentReleaseVersion.Id);
+
+            result.AssertNotFound();
+        }
+
+        [Fact]
+        public async Task ListSubjects_StatsDbHasMissingSubject()
+        {
+            var statisticsReleaseVersion = new Data.Model.ReleaseVersion();
+
+            var releaseSubject1 = new ReleaseSubject
+            {
+                ReleaseVersion = statisticsReleaseVersion,
+                Subject = new Subject()
+            };
+
+            var subject1Filter1 = new Filter
+            {
+                Subject = releaseSubject1.Subject,
+                Label = "subject 1 filter 1",
+            };
+
+            var subject1IndicatorGroup1 = new IndicatorGroup
+            {
+                Subject = releaseSubject1.Subject,
+                Label = "subject 1 indicator group 1",
+                Indicators = new List<Indicator>
+                {
+                    new() { Label = "subject 1 indicator group 1 indicator 1" },
+                }
+            };
+
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                statisticsDbContext.ReleaseSubject.Add(releaseSubject1);
+                statisticsDbContext.Filter.Add(subject1Filter1);
+                statisticsDbContext.IndicatorGroup.Add(subject1IndicatorGroup1);
+                await statisticsDbContext.SaveChangesAsync();
+            }
+
+            var contentReleaseVersion = new ReleaseVersion
+            {
+                Id = statisticsReleaseVersion.Id,
+            };
+
+            var releaseFile1 = new ReleaseFile
+            {
+                ReleaseVersion = contentReleaseVersion,
+                Name = "Data set 1",
+                File = new File
+                {
+                    Filename = "data1.csv",
+                    ContentLength = 10240,
+                    Type = FileType.Data,
+                    SubjectId = releaseSubject1.Subject.Id
+                },
+                Summary = "Data set 1 guidance"
+            };
+
+            var releaseFile2 = new ReleaseFile
+            {
+                ReleaseVersion = contentReleaseVersion,
+                Name = "Data set 2",
+                File = new File
+                {
+                    Filename = "data2.csv",
+                    ContentLength = 20480,
+                    Type = FileType.Data,
+                    SubjectId = Guid.NewGuid(),
+                },
+                Summary = "Data set 2 guidance"
+            };
+
+            var import1 = new DataImport
+            {
+                File = releaseFile1.File,
+                Status = DataImportStatus.COMPLETE
+            };
+
+            var import2 = new DataImport
+            {
+                File = releaseFile2.File,
+                Status = DataImportStatus.COMPLETE
+            };
```

Comment:
Because I copied the test data from another test. I've changed it to use the test data generator now, although not sure it was worth the time switching, in my opinion.